src/common/secret.c

```diff
@@ -85,6 +88,17 @@ int is_kernel_secret(const char *key_name)
   serial = request_key("ceph", key_name, NULL, KEY_SPEC_USER_KEYRING);
   return serial != -1;
 }
+#else
+int set_kernel_secret(const char *secret, const char *key_name)
+{
+  return -EINVAL;
+}
+
+int is_kernel_secret(const char *key_name)
+{
+  return -EINVAL;
```

Comment:
this is wrong, `is_kernel_secret()` returns 1 on success, 0 otherwise. we do not check the return code to see if it's negative or not. BTW, is mount.ceph and/or krbd compilable on FreeBSD? they are the only two callers of this function.